### PR TITLE
DB-10101: avoid read/write contention on SYSBACKUPS file

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1480/

**Review by bklo94 — APPROVED** [review]

File: hbase_sql/src/main/java/com/splicemachine/hbase/BackupUtils.java [hunk 1]
```diff
@@ -234,6 +231,12 @@ public static boolean shouldCaptureIncrementalChanges(FileSystem fs,Path rootDir
                             }
                         }
                     }
+                    else if (BackupUtils.existsDatabaseBackup(fs, rootDir)) {
```
Comment (dgomezferro) [inline hbase_sql/src/main/java/com/splicemachine/hbase/BackupUtils.java:234]: What happens if there's a table/schema backup going on but there was a previous full backup ? Without this change we would register the HFile and with it we wouldn't. Should we change the check to `if (!shouldRegister && BackupUtils.existsDatabaseBackup(fs, rootDir)` ? Would that create a race condition in the SYSBACKUPS file too?

**Review by dgomezferro — CHANGES_REQUESTED** [review]

File: hbase_sql/src/main/java/com/splicemachine/hbase/BackupUtils.java (hunk 1 above)
Comment (jyuanca) [reply hbase_sql/src/main/java/com/splicemachine/hbase/BackupUtils.java]: Good point!. Table/schema backup should not write to SYSBACKUPS file. I made changes to ee branch.

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1481/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by jyuanca** [comment]
Please also review ee branch

**Review by martinrupp — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Review by dgomezferro — APPROVED** [review]

**Pull request merged**